### PR TITLE
syncer: fix location usage in #587

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
hotfix location usage in #587 

### What is changed and how it works?
1. use `location.Clone()`
2. use `currentLocation` for dml

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: syncer/syncer.go [hunk 1]
```diff
@@ -759,7 +759,7 @@ func (s *Syncer) addJob(job *job) error {
 				continue
 			}
 			for _, sourceTable := range tbs {
-				s.saveTablePoint(sourceSchema, sourceTable, job.location)
+				s.saveTablePoint(sourceSchema, sourceTable, job.location.Clone())
```
Comment (csuzhangxc) [inline syncer/syncer.go:762]: Is this `Clone` needed?

Comment (lichunzhu) [reply syncer/syncer.go:762]: It's not neccessary.

**Comment by lichunzhu** [comment]
This PR is already included in https://github.com/pingcap/dm/pull/625. Close this now.

**Pull request closed without merging**